File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.js
function solution(lines) {
    let answer = 0;
    const map = new Map();
    
    for (const line of lines) {
        for (let i = line[0]; i < line[1]; i++) {
            const key = `${i}, ${i + 1}`;
            map.set(key, (map.get(key) || 0) + 1);
        }
    }
    
    for (const count of map.values()) {
        if (count > 1) {
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<int>> lines) {
    int answer = 0;
    
    map<pair<int, int>, int> um;
    
    for(auto& line : lines){
        for(int i = line[0] ; i < line[1] ; i++)
            um[make_pair(i, i + 1)]++;
    }
    
    for(auto& m : um)
        if(m.second > 1)
            answer++;
    
    return answer;
}